src/WhatToCook.Application/Services/MealPlanningService.cs

```diff
@@ -38,26 +44,37 @@ public async Task<PlanOfMeals> Create(PlanOfMealRequest planOfMealRequest)
     public async Task<PlanOfMeals> Update(UpdatePlanOfMealRequest planOfMealRequest)
     {
         var mealPlanToUpdate = await _mealPlanningRepository.GetMealPlanByName(planOfMealRequest.Name);
-        var RecipeForMealPlanUpdate = planOfMealRequest.Recipes.Select(x => x.Name);
 
-        var recipes = _recipesRepository.GetRecipesByNameForMealPlan(RecipeForMealPlanUpdate);
         if (mealPlanToUpdate == null)
         {
             _logger.LogError($"Attempted to update a non-existent meal plan: {planOfMealRequest.Name}");
             throw new NotFoundException(nameof(mealPlanToUpdate));
         }
-        //check if all recipes in the request exist in the database
-        var existingRecipes = recipes.Select(r => r.Name).ToList();
-        if (!existingRecipes.OrderBy(n => n).SequenceEqual(RecipeForMealPlanUpdate.OrderBy(n => n)))
+
+        var RecipeForMealPlanUpdate = planOfMealRequest.Recipes.SelectMany(x => x.RecipeIds);
+        var recipes = _recipesRepository.GetRecipesByIdForMealPlan(RecipeForMealPlanUpdate);
+
+
+        var existingRecipes = recipes.Select(r => r.Id).ToList();
+
+        var missingRecipes = RecipeForMealPlanUpdate.Except(existingRecipes).ToList();
+        if (missingRecipes.Any())
         {
-            _logger.LogError($"Some recipes do not exist in the database");
-            throw new NotFoundException($"Some recipes do not exist in the database");
+            _logger.LogError($"Some recipes do not exist in the database. Missing IDs: {string.Join(", ", missingRecipes)}");
+            throw new NotFoundException($"Some recipes do not exist in the database. Missing IDs: {string.Join(", ", missingRecipes)}");
         }
 
         mealPlanToUpdate.Name = planOfMealRequest.Name;
         mealPlanToUpdate.SetDates(planOfMealRequest.FromDate, planOfMealRequest.ToDate);
-        mealPlanToUpdate.Recipes = recipes;;
+
+        var updatedRecipePlans = planOfMealRequest.Recipes
+        .SelectMany(dayRecipe => dayRecipe.RecipeIds, (dayRecipe, recipeId) =>
+        new RecipePlanOfMeals(recipes.First(r => r.Id == recipeId), mealPlanToUpdate, dayRecipe.Day)).ToList();
+
+        mealPlanToUpdate.RecipePlanOfMeals = updatedRecipePlans;
```

Comment:
I think we should not have possibility to freely replace data of domain object. I think it should be encapsulated